### PR TITLE
fix(analyzers): Analyzers may not show the proper status using the reference API

### PR DESCRIPTION
**Pull request by jeromelaban**
GitHub Issue (If applicable): fixes https://github.com/unoplatform/uno/issues/3905

## PR Type

What kind of change does this PR introduce?
- Bugfix

## What is the new behavior?

APIs defined as not implemented partially in netstandard reference APIs are now properly reported as not implemented in libraries.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.Analyzers/UnoNotImplementedAnalyzer.cs [hunk 1]
```diff
@@ -78,9 +82,63 @@ private void OnObjectCreationExpression(SyntaxNodeAnalysisContext contextAnalysi
...
+			var directives = contextAnalysis.Node.GetLocation()?.SourceTree.Options.PreprocessorSymbolNames.ToArray() ?? new string[0];
+
+			if (directives.Length == 0)
+			{
+				var directive = contextAnalysis
+					.Node
+					.GetLocation()
+					.SourceTree
+					.GetRoot()
+					.GetFirstDirective() as DefineDirectiveTriviaSyntax;
```
Comment (davidjohnoliver) [inline src/Uno.Analyzers/UnoNotImplementedAnalyzer.cs:96]: I'm curious what this will do?

Comment (jeromelaban) [reply src/Uno.Analyzers/UnoNotImplementedAnalyzer.cs:96]: I'll document. It's only used during tests, as in normal compilations, there are (almost) always symbols.

**Comment by jeromelaban** [comment]
@mergifyio backport release/stable/3.0

**Pull request merged**